paho/client_test.go

```diff
@@ -353,6 +366,80 @@ func TestClientReceiveQoS2(t *testing.T) {
 	<-rChan
 }
 
+func TestClientReceiveAndAckInOrder(t *testing.T) {
```

Comment:
I used a slightly different approach in this test, I hope you don't mind.
Instead of starting routines manually and initialize unexported parts of the clients, I'm initializing the client in a way that is closer to how a user would do it (given that they can't just do `c.stop = make(chan struct{})`). I find it less error prone, let me know what you think.